cas_implementations/src/cas/file.rs

```diff
@@ -1,78 +1,77 @@
-pub mod actor;
-use actor::{AskSelf, Protocol};
-use cas::file::actor::FilesystemStorageActor;
 use holochain_core_types::{
     cas::{
         content::{Address, AddressableContent, Content},
         storage::ContentAddressableStorage,
     },
     error::HolochainError,
 };
-use riker::actors::*;
+use std::{
+    fs::{create_dir_all, read_to_string, write},
+    path::{Path, MAIN_SEPARATOR},
+    sync::{Arc, RwLock},
+};
 
 use uuid::Uuid;
 
-#[derive(Clone, PartialEq, Debug)]
+#[derive(Clone, Debug)]
 pub struct FilesystemStorage {
-    actor: ActorRef<Protocol>,
+    /// path to the directory where content will be saved to disk
+    dir_path: String,
     id: Uuid,
+    lock: Arc<RwLock<()>>,
```

Comment:
This looks great so far more lightweight! I was wondering if we need the extra guard since we are using this behind the ARC and Mutex

Reply:
Hm, you're saying because we have the storage wrapped in `Arc<RwLock<>>` in the context already we don't even need this lock here? You're maybe right...

Reply:
Oh yeah! Cause the writes will be around that lock so you can't do a write and read without unwrapping which in order to do that frees it from it's lock :)